JunList.java

```diff
@@ -19,99 +21,104 @@ public void append(int value) {
             int[] newList = new int[list.length * 2];
             for (int i = 0; i < list.length; i++) {
                 newList[i] = this.list[i];
-                this.list = newList;
             }
-            this.list[trackListPosition] = value;
-            trackListPosition += 1;
+            this.list = newList;
         }
+        this.list[trackListPosition] = value;
+        trackListPosition += 1;
+    }
 
-        public void add(int position,int value){

```

Comment:
When I test this code manually, I insert 10k elements then check the size and some elements, it works. Then I removed 10k elements and checked the size, correct in 0. However I was able to get 11 as a value from the get method. Funny enough I wasn't trying to do this, as I was just trying to remove the first element which worked correctly, but when I ended up removing elements outside of the list inside of the internal arrays you allowed me to remove them but technically they didn't exist. I should've gotten an exception or some way of not allowing it.

Reply:
Here's the code I used:
`        JunList list = new JunList();

        for (int i = 0; i < 10000; i++) {
            list.append(i);
        }
        System.out.println(list.size());
        for (int i = 0; i < 10000; i++) {
            list.remove(i);
        }
        System.out.println(list.size());
        System.out.println(list.get(5));
    }`

Reply:
Output:
10000
0
11

Reply:
If you're confused by what I mean, after I remove 10k elements from your list of 10k elements,
1. You shouldn't have allowed me to because I was accidentally removing elements from indexes that weren't there and I was accessing elements that should've been considered out of bounds.
2. Meaning this code was written to do the happy path correctly, but my dumb ass should've broken it but I didn't.